lib/os/shared_multi_heap.c

```diff
@@ -8,43 +8,31 @@
 #include <device.h>
 #include <sys/sys_heap.h>
 #include <sys/multi_heap.h>
-#include <linker/linker-defs.h>
 
 #include <multi_heap/shared_multi_heap.h>
 
-#define DT_DRV_COMPAT shared_multi_heap
-
-#define NUM_REGIONS DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)
-
 static struct sys_multi_heap shared_multi_heap;
-static struct sys_heap heap_pool[SMH_REG_ATTR_NUM][NUM_REGIONS];
-
-static smh_init_reg_fn_t smh_init_reg;
-
-#define FOREACH_REG(n)								\
-	{ .addr = (uintptr_t) LINKER_DT_RESERVED_MEM_GET_PTR(DT_DRV_INST(n)),	\
-	  .size = LINKER_DT_RESERVED_MEM_GET_SIZE(DT_DRV_INST(n)),		\
-	  .attr = DT_ENUM_IDX(DT_DRV_INST(n), capability),			\
-	},
+static struct sys_heap heap_pool[MAX_SHARED_MULTI_HEAP_ATTR][MAX_MULTI_HEAPS];
 
-static struct shared_multi_heap_region dt_region[NUM_REGIONS] = {
-	DT_INST_FOREACH_STATUS_OKAY(FOREACH_REG)
-};
+static unsigned int attr_cnt[MAX_SHARED_MULTI_HEAP_ATTR];
 
 static void *smh_choice(struct sys_multi_heap *mheap, void *cfg, size_t align, size_t size)
 {
-	enum smh_reg_attr attr;
 	struct sys_heap *h;
+	unsigned int attr;
 	void *block;
 
-	attr = (enum smh_reg_attr) cfg;
+	attr = (unsigned int)(long) cfg;
```

Comment:
Can I ask why not `attr = (unsigned int) cfg`

Reply:
because the compiler will complain otherwise

Reply:
Got it. something I don't know